### PR TITLE
All task properties in the updatable Task view

### PR DESCRIPTION
**Pull request by dbarashev**
So now we have an updatable view "Task" in the server-side PostgreSQL database and its schema matches"Task" table in the client-side in-memory DB, except "uid" column which is not yet in H2. It means that SQL statement from the log can be applied on the server side.

### CODE REVIEW

**Comment by dbarashev** [comment]
Update #2096

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql [hunk 1]
```diff
@@ -208,39 +231,126 @@ CREATE TABLE TaskName(
...
+       num,
+       name,
+       start_date,
+       duration_days AS duration,
+       earliest_start_date,
+       is_cost_calculated,
+       cost_manual_value,
+       is_milestone,
+       is_project_task,
+       MAX(TIP.prop_value) FILTER (WHERE TIP.prop_name = 'completion') AS completion,
```
Comment (npostnikova) [inline cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql:282]: Max is nominal here to confirm that we select a single value, right?

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql [hunk 2]
```diff
@@ -208,39 +231,126 @@ CREATE TABLE TaskName(
...
+    INSERT INTO project_template.TaskTextProperties(uid, prop_name, prop_value) VALUES(task_uid, prop_name_, prop_value_)
+    ON CONFLICT(uid, prop_name) DO UPDATE SET prop_value=prop_value_;
+$$;
 
 CREATE OR REPLACE FUNCTION update_task_row() RETURNS TRIGGER AS $$
 BEGIN
-INSERT INTO TaskName(uid, num, name)
-SELECT NEW.uid, NEW.num, NEW.name
-ON CONFLICT(uid) DO UPDATE SET num=NEW.num, name=NEW.name;
+INSERT INTO TaskName(uid, num, name) VALUES (NEW.uid, NEW.num, NEW.name)
```
Comment (npostnikova) [inline cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql:310]: Indent here would be nice

**Review by npostnikova — APPROVED** [review]
Great!

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql [hunk 3]
```diff
@@ -208,39 +231,126 @@ CREATE TABLE TaskName(
...
 -- Updatable view which collects all task properties in a single row. Inserts and updates are processed
 -- with INSTEAD OF triggers.
 CREATE VIEW Task AS
-SELECT uid, num, name, start_date, duration_days AS duration,
-       MAX(prop_value) FILTER(WHERE prop_name = 'completion') AS completion
-from TaskName JOIN TaskDates USING(uid)
-         LEFT JOIN TaskIntProperties USING(uid)
-GROUP BY uid, TaskDates.uid;
+SELECT uid,
+       num,
```
[the remark this answers is not in the document]
Comment (dbarashev) [reply cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql:273]: yes, let's rename it. `id` carries uniqueness semantics

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql (hunk 1 above)
Comment (dbarashev) [reply cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql:282]: Yes, just to make SELECT+GROUP BY happy. In a group filtered by the property name we'll have no more than 1 value anyway.

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql (hunk 2 above)
Comment (dbarashev) [reply cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql]: done

File: cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql (hunk 1 above)
Comment (npostnikova) [reply cloud.ganttproject.colloboque/src/main/resources/init-colloboque-template-database.sql:282]: Alright, thanks!

**Pull request merged**